x-pack/plugins/file_upload/public/components/geojson_upload_form/geojson_upload_form.tsx

```diff
@@ -80,7 +56,7 @@ export class GeoJsonUploadForm extends Component<Props, State> {
 
     this.props.onFileSelect(onFileSelectParameters);
 
-    this._onIndexNameChange(onFileSelectParameters.indexName);
+    this.props.onIndexNameChange(onFileSelectParameters.indexName);
```

Comment:
You will still need to check the new name to make sure there are no error and also propagate errors. Otherwise a user could open a file with a duplicated index name or illegal characters. To make this easier, I think it would be best to remove loading of index names and instead use the `/internal/file_upload/index_exists` route that was just migrated over from ml.

Reply:
It still does all the error checking you're referring to. I ran it with the latest changes before pushing the last update. The latest index name gets pushed down to the `IndexNameForm` where it's error checked. After error checking it executes the callback with the latest index name (which hasn't changed) and the associated error.

Reply:
Doing the check by propagating index name up to JsonUploadAndParse and then pushing the index name to  IndexNameForm and having IndexNameForm.componentDidUpdate perform the check is really hacky and does not follow typical react flows. It would be best if checking the index name is a stand alone function that can be called anytime the index name changes and is not tied to a react component.

Reply:
Anyone who wants to use this component would want these checks wouldn't they? To me it makes sense to bundle them together. Would you envision the error checking logic being a util function that we provide separately from the file upload start contract to any consumer?

Reply:
> Would you envision the error checking logic being a util function that we provide separately from the file upload start contract to any consumer?

Yes. So when using the component, the component will perform the error checking before calling onChange. If the index name is modified outside of the component, then error checking can happen but is not couple to the component. For example, consumers may want to ensure the index name is valid but not use the IndexNameComponent.

Reply:
Code updates:
- Created a separate validation function in utils that can be called from both places
- Converted the services to TS
- Used the index pattern service instead of the home-grown option we were using
- Stuck with the original index name check. I think it makes sense to retain this for the purposes of this PR